lib/elixir/lib/integer.ex

```diff
@@ -225,12 +254,12 @@ defmodule Integer do
 
   """
   @spec to_string(integer, 2..36) :: String.t
-  def to_string(some_integer, base) do
-    :erlang.integer_to_binary(some_integer, base)
+  def to_string(integer, base) when is_integer(integer) and is_integer(base) and base in 2..36 do
```

Comment:
Actually, I checked exactly that with the compiled version before asking, and this is what we get:
```
iex(1)> Integer.to_string("foo")
** (ArgumentError) argument error
    :erlang.integer_to_binary("foo")
```
and

```elixir 
iex(1)> &Integer.to_string/1     
&:erlang.integer_to_binary/1
```
Doing that told me that guards applied to functions inlined by the compiler have no effect,
I think it's because of the rewrite rules.
https://github.com/elixir-lang/elixir/blob/master/lib/elixir/src/elixir_rewrite.erl#L35-L36

isn't it?